lib/govuk_index/presenters/details_presenter.rb

```diff
@@ -1,48 +1,32 @@
 module GovukIndex
   class DetailsPresenter
-    DEFAULT_FIELDS = %w(body parts).freeze
-    FIELDS_BY_FORMAT = {
-      'licence'     => %w(licence_short_description licence_overview),
-      'transaction' => %w(introductory_paragraph more_information)
-    }.freeze
+    extend MethodBuilder
 
-    def initialize(details:, format:, sanitiser:)
+    set_payload_method :details
+
+    delegate_to_payload :licence_identifier
+    delegate_to_payload :licence_short_description
+
+    def initialize(details:, indexable_content_keys:, sanitiser:)
       @details = details
-      @format = format
+      @indexable_content_keys = indexable_content_keys
       @sanitiser = sanitiser
     end
 
     def indexable_content
       return nil if details.nil?
-      sanitiser.clean(elements)
-    end
-
-    def licence_identifier
-      details['licence_identifier']
-    end
-
-    def licence_short_description
-      details['licence_short_description']
+      @sanitiser.clean(indexable_content_parts)
     end
 
   private
 
-    attr_reader :details, :format, :sanitiser
+    attr_reader :details
```

Comment:
Might as well remove this attr_reader too and use the escargot.

Reply:
can't as it is accessed from the MethodBuidler code, I remove the other two as the accessor is only used in a single location, so it seem overkill.